test/Microsoft.SqlTools.ServiceLayer.TestDriver/Tests/LanguageServiceTests.cs

```diff
@@ -279,5 +279,137 @@ public class LanguageServiceTests : TestBase
                 WaitForExit();
             }
         }
+
+        [Fact]
+        public async Task FunctionSignatureCompletionReturnsCorrectFunction()
+        {
+            string sqlText = "EXEC sys.fn_isrolemember ";
+            string ownerUri = System.IO.Path.GetTempFileName();
+
+            try
+            {
+                System.IO.File.WriteAllText(ownerUri, sqlText);
+
+                // Connect
+                await Connect(ownerUri, ConnectionTestUtils.LocalhostConnection);
+
+                // Wait for intellisense to be ready
+                var readyParams = await Driver.WaitForEvent(IntelliSenseReadyNotification.Type, 30000);
+                Assert.NotNull(readyParams);
+                Assert.Equal(ownerUri, readyParams.OwnerUri);
+
+                // Send a function signature help Request
+                var position = new TextDocumentPosition()
+                {
+                    TextDocument = new TextDocumentIdentifier()
+                    {
+                        Uri = ownerUri
+                    },
+                    Position = new Position()
+                    {
+                        Line = 0,
+                        Character = sqlText.Length
+                    }
+                };
+                var signatureHelp = await Driver.SendRequest(SignatureHelpRequest.Type, position);
+
+                Assert.NotNull(signatureHelp);
+                Assert.True(signatureHelp.ActiveSignature.HasValue);
+                Assert.NotEmpty(signatureHelp.Signatures);
+
+                var label = signatureHelp.Signatures[signatureHelp.ActiveSignature.Value].Label;
+                Assert.NotNull(label);
+                Assert.NotEmpty(label);
+                Assert.True(label.Contains("fn_isrolemember"));
+
+                await Disconnect(ownerUri);
+            }
+            finally
+            {
+                System.IO.File.Delete(ownerUri);
+                WaitForExit();
+            }
+        }
+
+        [Fact]
+        public async Task FunctionSignatureCompletionReturnsCorrectParametersAtEachPosition()
```

Comment:
We should have some negative test, e..g for when the location.Name doesn't match. Could you add this one?